pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -148,19 +148,29 @@ void   _array_fill_cdouble(double complex c, t_ndarray arr)
 
 int32_t free_array(t_ndarray dump)
 {
-    if (!dump.is_slice)
-    {
-        free(dump.raw_data);
-        dump.raw_data = NULL;
-    }
+    if (dump.shape == NULL)
+        return (0);
+    free(dump.raw_data);
+    dump.raw_data = NULL;
     free(dump.shape);
     dump.shape = NULL;
     free(dump.strides);
     dump.strides = NULL;
     return (1);
 }

```

Comment:
do you use returns values somewhere in the code, or are planning to implement some error management later?

Reply:
i think it might be useful to anyone that want to make changes to the generated C code